shared/src/main/java/com/couchbase/lite/Query.java

```diff
@@ -170,7 +175,9 @@ public ResultSet execute() throws CouchbaseLiteException {
     public String explain() throws CouchbaseLiteException {
         if (c4query == null)
```

Comment:
c4query should also be synchronized. Move c4query checks to the check method itself.

The same thing for the execute() method.

See https://github.com/couchbase/couchbase-lite-ios/blob/feature/2.0/Objective-C/CBLQuery.mm#L605-L606